### PR TITLE
feat: wipe MLS group on conversation deletion WPB-543

### PR DESCRIPTION
**Pull request by David-Henner**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-543" title="WPB-543" target="_blank">StoryWPB-543</a>  [iOS] Wipe local MLS group when a conversation gets deleted
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

When a conversation is deleted, the corresponding MLS group needs to be wiped 

### Changes

- Created `LocalConversationRemovalUseCase` to mark conversation as deleted and wipe the MLS group if the protocol is `mls`
- Used the use case when deleting conversations

### Testing

- [x] Added unit tests

### CODE REVIEW

**Review by agisilaos — APPROVED** [review]

File: wire-ios-data-model/Use cases/LocalConversationRemovalUseCase.swift [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+//
+
+import Foundation
+import WireSystem
+
+public protocol LocalConversationRemovalUseCaseProtocol {
+    func removeConversation(_ conversation: ZMConversation, syncContext: NSManagedObjectContext)
+}
+
+public class LocalConversationRemovalUseCase: LocalConversationRemovalUseCaseProtocol {
```
Comment (netbe) [inline wire-ios-data-model/Use cases/LocalConversationRemovalUseCase.swift]: talking about useCases in another PR, we should name them with an action verb,

so RemoveLocalConversationUseCase

File: wire-ios-data-model/Use cases/LocalConversationRemovalUseCase.swift [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    public init() {}
+
+    public func removeConversation(
```
Comment (netbe) [inline wire-ios-data-model/Use cases/LocalConversationRemovalUseCase.swift]: As the use case does only one action of its name we can just name it with less context
```suggestion
    public func invoke(
```

File: wire-ios-request-strategy/Sources/Payloads/Processing/ConversationEventPayloadProcessor.swift [hunk 3]
```diff
@@ -25,6 +26,12 @@ final class ConversationEventPayloadProcessor {
         case eventStream
     }
 
+    private let localConversationRemovalUseCase: LocalConversationRemovalUseCaseProtocol
```
Comment (netbe) [inline wire-ios-request-strategy/Sources/Payloads/Processing/ConversationEventPayloadProcessor.swift]: we also drop the UseCase suffix for properties
```suggestion
    private let removeLocalConversation: LocalConversationRemovalUseCaseProtocol
```

**Review by netbe — CHANGES_REQUESTED** [review]
Here's some inputs to be consistent with other UseCases

File: wire-ios-data-model/Use cases/LocalConversationRemovalUseCase.swift (hunk 1 above)
Comment (David-Henner) [reply wire-ios-data-model/Use cases/LocalConversationRemovalUseCase.swift]: @netbe could you link the PR?

Comment (netbe) [reply wire-ios-data-model/Use cases/LocalConversationRemovalUseCase.swift]: https://github.com/wireapp/wire-ios/pull/615#discussion_r1369886241

File: wire-ios-request-strategy/Sources/Payloads/Processing/ConversationEventPayloadProcessor.swift (hunk 3 above)
Comment (David-Henner) [reply wire-ios-request-strategy/Sources/Payloads/Processing/ConversationEventPayloadProcessor.swift]: done

File: wire-ios-data-model/Use cases/LocalConversationRemovalUseCase.swift (hunk 2 above)
Comment (David-Henner) [reply wire-ios-data-model/Use cases/LocalConversationRemovalUseCase.swift]: done

File: wire-ios-data-model/Use cases/LocalConversationRemovalUseCase.swift (hunk 1 above)
Comment (David-Henner) [reply wire-ios-data-model/Use cases/LocalConversationRemovalUseCase.swift]: done

**Review by netbe — APPROVED** [review]

**Pull request merged**